core/src/duplicate_shred.rs

```diff
@@ -0,0 +1,338 @@
+use crate::crds_value::sanitize_wallclock;
+use itertools::Itertools;
+use solana_ledger::{
+    blockstore_meta::DuplicateSlotProof,
+    shred::{Shred, ShredError, ShredType},
+};
+use solana_sdk::{
+    clock::Slot,
+    pubkey::Pubkey,
+    sanitize::{Sanitize, SanitizeError},
+};
+use std::{
+    collections::{hash_map::Entry, HashMap},
+    convert::TryFrom,
+};
+use thiserror::Error;
+
+const DUPLICATE_SHRED_HEADER_SIZE: usize = 63;
+
+#[derive(Clone, Debug, PartialEq, AbiExample, Deserialize, Serialize)]
+pub struct DuplicateShred {
```

Comment:
nit: Rename to DuplicateShredChunk

Reply:
Leaning on brevity here, since the chunks are an implementation detail hidden inside the struct (i.e. they are not exposed outside this file). We can revisit later if there was ambiguity.